### PR TITLE
perf(build): make maxperf-op

### PR DESCRIPTION
**Pull request by emhane**
Adds make command for building op-reth maxperf

### CODE REVIEW

**Review by shekhirin — CHANGES_REQUESTED** [review]
I think we should rather have a separate makefile target for that, because the current change sets the expectation for having a nightly toolchain.

File: Makefile [hunk 1]
```diff
@@ -261,11 +261,15 @@ update-book-cli: ## Update book cli documentation.
 
 .PHONY: maxperf
 maxperf: ## Builds `reth` with the most aggressive optimisations.
-	RUSTFLAGS="-C target-cpu=native" cargo build --profile maxperf --features jemalloc,asm-keccak
+	RUSTFLAGS="-C target-cpu=native -Z threads=8" cargo +nightly build --profile maxperf --features jemalloc,asm-keccak
```
Comment (shekhirin) [inline Makefile:264]:
```suggestion
	RUSTFLAGS="-C target-cpu=native -Z threads=$(nproc)" cargo +nightly build --profile maxperf --features jemalloc,asm-keccak
```

**Comment by onbjerg** [comment]
just to note here, the reason we do not do this is because the only thing that is currently parallel in rustc is codegen (https://rustc-dev-guide.rust-lang.org/parallel-rustc.html#parallel-compilation), and codegenning with more than 1 thread may produce slower code: https://doc.rust-lang.org/cargo/reference/profiles.html#codegen-units

File: Makefile (hunk 1 above)
Comment (emhane) [reply Makefile:264]: oo, cool

Comment (emhane) [reply Makefile:264]: but if you don't have the command installed it's not so cool. besides, don't think anyone with less cores builds maxperf tbh.

**Comment by emhane** [comment]
> just to note here, the reason we do not do this is because one of the only thing that is currently parallel in rustc is codegen (https://rustc-dev-guide.rust-lang.org/parallel-rustc.html#parallel-compilation), and codegenning with more than 1 thread may produce slower code: https://doc.rust-lang.org/cargo/reference/profiles.html#codegen-units
> 
> as such, this should be a different makefile target - we do not want to use this for builds in our releases
> 
> we explicitly set codgen units to 1 for maxperf:
> 
> https://github.com/paradigmxyz/reth/blob/12873d515a9cea30d553fe938dc42a12c072562b/Cargo.toml#L200-L203

wow no way, didn't know parallelisation came at price of optimisation for rustc

File: Makefile (hunk 1 above)
Comment (shekhirin) [reply Makefile]: iirc it's a unix command, so everyone should have it. you can additionally do ` || 8`

Comment (Rjected) [reply Makefile]: not available on macos

Comment (shekhirin) [reply Makefile]: > not available on macos

insane

File: Makefile [hunk 2]
```diff
@@ -263,6 +263,10 @@ update-book-cli: ## Update book cli documentation.
 maxperf: ## Builds `reth` with the most aggressive optimisations.
 	RUSTFLAGS="-C target-cpu=native" cargo build --profile maxperf --features jemalloc,asm-keccak
 
+.PHONY: maxperf-op
+maxperf-op: ## Builds `reth` with the most aggressive optimisations.
```
Comment (shekhirin) [inline Makefile:267]: comment needs an update

**Review by shekhirin — APPROVED** [review]
comment nit, otherwise LGTM

File: Makefile (hunk 2 above)
Comment (emhane) [reply Makefile:267]: more than a nit I would say 🫡

**Pull request merged**